lib/msal-browser/src/interaction_handler/PopupHandler.ts

```diff
@@ -111,25 +111,19 @@ export class PopupHandler extends InteractionHandler {
      * @ignore
      * @hidden
      */
-    private openPopup(urlNavigate: string, title: string, popUpWidth: number, popUpHeight: number): Window {
+    private openPopup(urlNavigate: string, popup?: Window|null): Window {
         try {
-            /**
-             * adding winLeft and winTop to account for dual monitor
-             * using screenLeft and screenTop for IE8 and earlier
-             */
-            const winLeft = window.screenLeft ? window.screenLeft : window.screenX;
-            const winTop = window.screenTop ? window.screenTop : window.screenY;
-            /**
-             * window.innerWidth displays browser window"s height and width excluding toolbars
-             * using document.documentElement.clientWidth for IE8 and earlier
-             */
-            const width = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
-            const height = window.innerHeight || document.documentElement.clientHeight || document.body.clientHeight;
-            const left = Math.max(0, ((width / 2) - (popUpWidth / 2)) + winLeft);
-            const top = Math.max(0, ((height / 2) - (popUpHeight / 2)) + winTop);
-
-            // open the window
-            const popupWindow = window.open(urlNavigate, title, "width=" + popUpWidth + ", height=" + popUpHeight + ", top=" + top + ", left=" + left);
+            let popupWindow;
+            // Popup window passed in, setting url to navigate to
+            if (popup) {
```

Comment:
Out of curiosity what's the reason why we open a blank popup first by default? Why would we not want to change the code to have a single code path where we just always open the popup directly to the /authorize endpoint?

Reply:
It appears that a number of browsers will block popups that are opened in asynchronous code paths even if they wouldn't otherwise be blocked (e.g. as a result of user interaction). The mitigation is to synchronously open a blank popup and asynchronously update it with the url. The issue reported has more information on this: #2001

Reply:
Sorry, guess I misunderstood the behavior today vs. the fix. My question was more around, if that's the case why don't we always open a blank popup synchronously and update asynchronously with the url? What's the reasoning behind creating an additional code path and parameter vs. just changing the existing behavior to fix the issue on those browsers? Would that break the other browsers that are currently not broken?

Reply:
Oh, I see what you're asking. The fix doesn't break with other browsers, so we are having it as the default code path. However, it seems that some desktop apps and progressive web apps have problems with blank popups, so in msal-core we had to stop opening blank popups entirely and open them later with the url. Having two code paths in this fix tries make sure that both kinds of issues are covered. Here's the other PR and related issues for desktop apps and PWA: #1667

Reply:
Oh that's right, makes sense. Thanks for clearing that up!